kernel/src/mem.rs

```diff
@@ -103,32 +120,43 @@ impl<L, T> AppSlice<L, T> {
             false
         }
     }
+}
 
-    pub fn iter(&self) -> slice::Iter<T> {
-        self.as_ref().iter()
+impl<L: Read, T> AppSlice<L, T> {
+    pub fn make_read_only(self) -> AppSlice<SharedRO, T> {
```

Comment:
Does this mean that you can create multiple Read AppSlices from a single AppSlice?

Suppose I have an API that takes an AppSlice<SharedRO>. But a capsule wants to use an AppSlice<Shared>. It'll then call `make_read_only()` to generate an AppSlice<SharedRO>. It will hold onto its AppSlice<Shared> if it later needs to use it as read-write.This is valid, right? For example, an app could also erroneously pass the same memory to multiple `allow` calls.

Reply:
No, `make_read_only` takes `self` not `&self`, so it consumes it's argument. This merely allows you to convert from any `AppSlice` (because all variants of `L` implement `Read`) to a read-only AppSlice.

It's useful in the case that we have a backwards compatible API for a driver that uses a regular allow where only a read-only slice is necessary.